src/vm.rs

```diff
@@ -651,13 +660,14 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
                 return Err(EcallError::from(SbiError::InvalidParam));
             }
             active_vcpu.is_counter_range_startable(counter_index, counter_mask)?;
-            pmu::start_counters(counter_index, counter_mask, start_flags, initial_value)?;
-            active_vcpu.update_started_counters(
-                counter_index,
-                counter_mask & hardware_counters_mask,
-                start_flags,
-                initial_value,
-            );
+            let result =
+                pmu::start_counters(counter_index, counter_mask, start_flags, initial_value);
+            // Special case "already started" to handle counters that are autostarted following configuration
```

Comment:
Can you be more specific? I'm guessing you're referring to the cycle and instret counters...?

Reply:
Yes, but it seems really odd to special case it, and we don't know if other platforms might have similar exceptions.

Reply:
The exceptions for cycle and instret are going to be common across RISC-V platforms I would think. I'm just suggesting that this comment be more specific about where these special cases arise e.g. "The legacy `cycle` and `instret` counters may be auto-started following configuration".

Reply:
Auto start can happen for any counters. it doesn't have to be cycle/instret. Auto start parameter was added where supervisor prefers not to change the mapping once obtained. In that case, it can save an SBI call for starting.